uploader/s3.go

```diff
@@ -87,6 +90,11 @@ func (u *S3Uploader) Upload(ctx context.Context, local string, remote string, ct
 		}
 	}()
 
+	// Record number of bytes being uploaded.  At this point, file must exist.
+	if stat, err := os.Stat(local); err == nil {
```

Comment:
Don't use os.Stat here because then you wont catch stuff being uploaded via `UploadPartOfFile`.

You should move the code into `uploadFile`. I might see if there's a simple io.Reader wrapper to get callbacks as data is read.

Reply:
fixed